core/src/main/java/org/apache/iceberg/rest/RESTTableOperations.java

```diff
@@ -241,12 +328,38 @@ private static Long expectedSnapshotIdIfSnapshotAddOnly(List<MetadataUpdate> upd
     return addedSnapshotId;
   }
 
+  private void encryptionPropsFromMetadata(TableMetadata metadata) {
+    if (metadata == null || metadata.properties() == null) {
+      return;
+    }
+
+    encryptedKeysFromMetadata = metadata.encryptionKeys();
+
+    Map<String, String> tableProperties = metadata.properties();
+    if (tableKeyId == null) {
```

Comment:
Shall we always get `tableKeyId` in case there is a key rotation?

Reply:
if the `tableKeyId == null` (here and below), it just means the table is not encrypted.